plugins/outputs/nats/nats.go

```diff
@@ -324,6 +401,30 @@ func (n *NATS) Write(metrics []telegraf.Metric) error {
 	return nil
 }
 
+func isSubjectDynamic(tpl template.Template, subject string) bool {
+	var buf bytes.Buffer
+	err := tpl.Execute(&buf, nil)
+	if err != nil || buf.String() != subject {
+		return true
+	}
+	return false
+}
```

Comment:
Please merge this function on the caller side! You might also want to use [`io.Discard`](https://pkg.go.dev/io#Discard) as writer because we don't need the actual result.